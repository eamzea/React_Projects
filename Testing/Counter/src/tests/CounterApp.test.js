import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en <CounterApp/>", () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("Debe mostrar <CounterApp/> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar value enviado por props", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("100");
  });

  test("Debe de incrementar con el botón +1", () => {
    const wrapper = shallow(<CounterApp value={10} />);

    wrapper.find(".buttonContainer").childAt(2).simulate("click");

    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("11");
  });

  test("Debe de incrementar con el botón -1", () => {
    const wrapper = shallow(<CounterApp value={10} />);

    wrapper.find(".buttonContainer").childAt(0).simulate("click");

    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("9");
  });

  test("Debe de regresar al valor inicial", () => {
    const wrapper = shallow(<CounterApp value={105} />);

    wrapper.find(".buttonContainer").childAt(2).simulate("click");
    wrapper.find(".buttonContainer").childAt(2).simulate("click");
    wrapper.find(".buttonContainer").childAt(1).simulate("click");

    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("105");
  });
});
