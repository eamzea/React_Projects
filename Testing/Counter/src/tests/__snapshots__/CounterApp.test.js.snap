// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp/> Debe mostrar <CounterApp/> correctamente 1`] = `
<div
  className="container"
>
  <h1>
    Counter App
  </h1>
  <h2 />
  <div
    className="buttonContainer"
  >
    <button
      onClick={[Function]}
    >
      - 1
    </button>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      onClick={[Function]}
    >
      + 1
    </button>
  </div>
</div>
`;
